rmw_zenoh_cpp/src/detail/rmw_subscription_data.cpp

```diff
@@ -173,151 +173,60 @@ bool SubscriptionData::init()
-    // by various publishers.
-    sub_options.query_consolidation =
-      zenoh::QueryConsolidation(zenoh::ConsolidationMode::Z_CONSOLIDATION_MODE_NONE);
-
-    std::weak_ptr<SubscriptionData> data_wp = shared_from_this();
-    auto sub = context_impl->session()->ext().declare_querying_subscriber(
-      sub_ke,
-      [data_wp](const zenoh::Sample & sample) {
-        auto sub_data = data_wp.lock();
-        if (sub_data == nullptr) {
-          RMW_ZENOH_LOG_ERROR_NAMED(
-            "rmw_zenoh_cpp",
-            "Unable to obtain SubscriptionData from data for %s.",
-            std::string(sample.get_keyexpr().as_string_view()).c_str());
-          return;
-        }
-
-        auto attachment = sample.get_attachment();
-        if (!attachment.has_value()) {
-          RMW_ZENOH_LOG_ERROR_NAMED(
-            "rmw_zenoh_cpp",
-            "Unable to obtain attachment")
-          return;
-        }
-
-        auto attachment_value = attachment.value();
-        AttachmentData attachment_data(attachment_value);
-
-        sub_data->add_new_message(
-          std::make_unique<SubscriptionData::Message>(
-            sample.get_payload(),
-            get_system_time_in_ns(),
-            std::move(attachment_data)),
-          std::string(sample.get_keyexpr().as_string_view()));
-      },
-      zenoh::closures::none,
-      std::move(sub_options),
-      &result);
-    if (result != Z_OK) {
-      RMW_SET_ERROR_MSG("unable to create zenoh subscription");
-      return false;
-    }
-    sub_ = std::move(sub);
-
-    // Register the querying subscriber with the graph cache to get latest
-    // messages from publishers that were discovered after their first publication.
-    graph_cache_->set_querying_subscriber_callback(
-      entity_->topic_info().value().topic_keyexpr_,
-      entity_->keyexpr_hash(),
-      [data_wp](const std::string & queryable_prefix) -> void
-      {
-        auto sub_data = data_wp.lock();
-        if (sub_data == nullptr) {
-          RMW_ZENOH_LOG_ERROR_NAMED(
-            "rmw_zenoh_cpp",
-            "Unable to lock weak_ptr<SubscriptionData> within querying subscription callback."
-          );
-          return;
-        }
-        std::lock_guard<std::mutex> lock(sub_data->mutex_);
-
-        const std::string selector = queryable_prefix +
-        "/" +
-        sub_data->entity_->topic_info().value().topic_keyexpr_;
-        RMW_ZENOH_LOG_DEBUG_NAMED(
+    adv_sub_opts.query_timeout_ms = std::numeric_limits<uint64_t>::max();
+    adv_sub_opts.history =
+      zenoh::ext::SessionExt::AdvancedSubscriberOptions::HistoryOptions::create_default();
+    adv_sub_opts.history->detect_late_publishers = true;
+    adv_sub_opts.history->max_samples = entity_->topic_info()->qos_.depth;
+    adv_sub_opts.recovery = zenoh::ext::SessionExt::AdvancedSubscriberOptions::RecoveryOptions{};
+    adv_sub_opts.recovery->periodic_queries_period_ms = 1000;
```

Comment:
Again looking at the [documentation](https://zenoh-cpp.readthedocs.io/en/latest/ext.html#_CPPv4N5zenoh3ext10SessionExt25AdvancedSubscriberOptions15RecoveryOptions26periodic_queries_period_msE), it sounds like recovery options should only be enabled if reliability is `RELIABLE`. Could you clarify on the behavior?

Reply:
Similarly, answered in https://github.com/ros2/rmw_zenoh/pull/368#discussion_r1912826044.

Reply:
Could we add some comments on why a value of `1000` is appropriate for `periodic_queries_period_ms`?